c7n/resources/directory.py

```diff
@@ -122,3 +151,49 @@ class resource_type(TypeInfo):
         universal_taggable = object()
 
     augment = universal_augment
+
+@CloudDirectory.action_registry.register('delete')
+class CloudDirectoryDelete(BaseAction):
+    """Disable/Delete a cloud directory.
+    
+    The directory(s) will be disabled when this action is called.
+    If force is True, the directory(s) will be permanently deleted
+
+    .. code-block:: yaml
+
+       policies:
+         - name: disable-cloud-directory
+           resource: aws.cloud-directory
+           filters:
+             - Name: test-cloud
+           actions:
+             - type: delete
+
+         - name: delete-cloud-directory
+           resource: aws.cloud-directory
+           filters:
+             - Name: test-cloud
+           actions:
+             - type: delete
+               force: True
+    """
+    schema = type_schema('delete', force={'type': 'boolean'})
+    permissions = ('clouddirectory:DeleteDirectory',
+                   'clouddirectory:DisableDirectory',)
+
+    def process(self, resources):
+        client = local_session(
+            self.manager.session_factory).client('clouddirectory')
+        force = self.data.get("force", False)
```

Comment:
Looks like custodian only filters the directories that are in `Enabled` state. Hence, we may not need a `force` option + `disablement` at all